src/dropdown/dropdown.ts

```diff
@@ -145,6 +146,13 @@ export class NgbDropdown implements OnInit {
       this._open = true;
       this._positionMenu();
       this.openChange.emit(true);
+      this._escapeListener = (event) => {
+        if (isEscape(event)) {
+          this.closeFromOutsideEsc();
+          this.changeDetector.detectChanges();
```

Comment:
hmm, why would you need to run the CD after closing?

Reply:
Honestly I can't remember why, probably it wasn't working properly otherwise, to be tested again.